compiler/lib/generate.ml

```diff
@@ -1132,6 +1132,7 @@ let rec translate_expr ctx queue loc _x e level : _ * J.statement_list =
                 ~init:([], mutator_p, queue)
             in
             ecall (J.EDot (co, m)) args loc, or_p po prop, queue
+        | Extern "%caml_js_opt_meth_call", _ :: Pc (String _) :: _ -> assert false
```

Comment:
The things commented out bellow were useful when trying to detect code that would call `Js.Unsafe.set x "ocaml-string" v` by mistake (instead of `Js.Unsafe.set x (Js.string "ocaml-string") v`).

`Js.Unsafe.set` has the signature `'a -> 'b -> 'c -> unit` so calling it with an ocaml string is valid per se, explaining why the assert is commented out.

Here, `%caml_js_opt_meth_call` is generated by the js_of_ocaml compiler. I want the assert to be compiled and checked.